game.py

```diff
@@ -89,12 +76,38 @@ def print_word_progress_string(snowman_word, snowman_word_dict):
         output_string += " "
     print(output_string)
 
-def get_word_progress(snowman_word, snowman_word_dict):
-    """
-    This function takes the snowman_word and snowman_word_dict as input.
-    It returns True if all the letters of the word have been guessed, and False otherwise.
-    """
-    for letter in snowman_word:
-        if not snowman_word_dict[letter]:
-            return False
-    return True
\ No newline at end of file
+
+def snowman(snowman_word):
+    r = RandomWord()
+    snowman_word = r.word(
+        word_min_length=SNOWMAN_MIN_WORD_LENGTH, 
+        word_max_length=SNOWMAN_MAX_WORD_LENGTH)
+
+    print(f"debug info: {snowman_word}")
+
+    correct_guesses_list = []
+    wrong_guesses_list = []
+    
+
+    while len(wrong_guesses_list) < SNOWMAN_WRONG_GUESSES:
+        user_input = get_letter_from_user(wrong_guesses_list, correct_guesses_list)
+        if user_input in snowman_word:
+            print("You guessed a letter that's in the word!")
+            correct_guesses_list.append(user_input)
+        else:
+                print("You've already guessed that letter correctly.")
+        
+        if user_input not in wrong_guesses_list:
+            wrong_guesses_list.append(user_input)
+            print(f"The letter {user_input} is not in the word")
+        else:
+            print("You've already guessed that letter incorrectly.")
+
+        print_snowman(len(wrong_guesses_list))
+        print(f"Wrong guesses: {wrong_guesses_list}")   
+        
+        if user_input and correct_guesses_list:
+            print("You win!")
+            return
+        print("Sorry, you lose!")
+        print(f"The word was '{snowman_word}'.")
```

Comment:
👀 These losing messages will print every time through the loop rather than once at the end of a lost game. If the game is lost, the loop will end, so these should be unindented so that they are after the loop.